internal/observability/graph.go

```diff
@@ -93,9 +93,9 @@ func newGraph(meter *metrics.Scope, logger *zap.Logger, extract ContextExtractor
 	}
 }
 
-// begin starts a call along an edge.
-func (g *graph) begin(ctx context.Context, rpcType transport.Type, direction directionName, req *transport.Request) call {
-	now := _timeNow()
+func (g *graph) begin(
+	ctx context.Context, rpcType transport.Type, direction directionName, req *transport.Request, c *call) {
```

Comment:
would returning a `*call` instead of `call` not be enough?

Reply:
It seems it is not the same. What you suggests escapes to the heap:

```
graph.go:113          0x59e3e                 e800000000              CALL 0x59e43                                                            [1:5]R_CALL:runtime.newobject
```

Reply:
I think one *call object escaping to heap is unavoidable - we might be able to over-fit this into our benchmark and temporarily avoid it, but it seems just like it's hard to avoid.

And if heap-ing means a significant code simplification, I'd choose that.